### ISSUE TITLE
max_bin is deprecated

### ISSUE THREAD

**Issue by QtRoS** [Microsoft/LightGBM#1100]
Something strange happened with `max_bin` parameter:

> .../anaconda3/lib/python3.6/site-packages/lightgbm/basic.py:642: UserWarning: max_bin keyword has been found in `params` and will be ignored. Please use max_bin argument of the Dataset constructor to pass this parameter.
>  'Please use {0} argument of the Dataset constructor to pass this parameter.'.format(key))
> .../anaconda3/lib/python3.6/site-packages/lightgbm/basic.py:648: LGBMDeprecationWarning: The `max_bin` parameter is deprecated and will be removed in 2.0.12 version. Please use `params` to pass this parameter.
>   'Please use `params` to pass this parameter.', LGBMDeprecationWarning)

I think that these messages are contradictory. 

## Environment info
Operating System: Ubuntu 17.10
CPU: AMD Ryzen 7 1800x
C++/Python/R version: Python 3.6
Version: '2.0.11'

**Comment by StrikerRUS** [comment]
Hi @QtRoS ! Please install the latest  version.

**Comment by QtRoS** [comment]
@StrikerRUS where I can find latest version? Tried conda, pip, installation from github.

**Comment by StrikerRUS** [comment]
@QtRoS GitHub master branch (https://github.com/Microsoft/LightGBM/commit/8a5ec366aa5b0ec6301cb6ca8e85a70cc7ac4cf1 and newer).

**Comment by dsun29** [comment]
@StrikerRUS  can you please share sample code to pass **max_bin** using **params**?

**Comment by StrikerRUS** [comment]
@dsun29 
```
import lightgbm as lgb

est = lgb.sklearn.LGBMModel(max_bin=32, objective='tweedie')
est.fit(your_X, your_y)
```

**Comment by dsun29** [comment]
@StrikerRUS
That is not what I was looking for bu thanks!
    `d_valid = lgb.Dataset(x_valid, label=y_valid, params = {'max_bin': p['lgb_max_bin']})`
